RELEASE_PROCESS.md

```diff
@@ -0,0 +1,64 @@
+# Our Release Process
+
+## Notes
+
+BigchainDB follows
+[the Python form of Semantic Versioning](https://packaging.python.org/tutorials/distributing-packages/#choosing-a-versioning-scheme)
+(i.e. MAJOR.MINOR.PATCH),
+which is almost identical
+to [regular semantic versioning](http://semver.org/), but there's no hyphen, e.g.
+
+- `0.9.0` for a typical final release
+- `4.5.2a1` not `4.5.2-a1` for the first Alpha release
+- `3.4.5rc2` not `3.4.5-rc2` for Release Candidate 2
+
+**Note 1:** For Git tags (which are used to identify releases on GitHub), we append a `v` in front. For example, the Git tag for version `2.0.0a1` was `v2.0.0a1`.
+
+**Note 2:** For Docker image tags (e.g. on Docker Hub), we use longer version names for Alpha, Beta and Release Candidate releases. For example, the Docker image tag for version `2.0.0a1` was `2.0.0-alpha`.
```

Comment:
This note might be confusing since AFAIK we don't make Docker images based on the Python driver. Maybe just delete this note.